greedy.py

```diff
@@ -1,25 +1,135 @@
 import ant_colony as ac
 import random
+import copy
 
 random.seed(0) # Ensure same random numbers every time
 
 # Perform one greedy step and return new game state (deep copy)
+# Greedy step rules:
+# 1. Every ant will head towards the closest piece of food to them
+# 2. If an enemy is 1 move away and food is 1 move away, attack enemy first
+# 3. Avoid enemy as much as possible while still advancing towards food
+# (maximize distance to enemy, minimize distance to food)
+# Ex. |1|-|-|     |-|1|-|
+#     |-|F|-| ->  |-|F|-|
+#     |2|-|-|     |2|-|-|
+# 4. When no more food left, ants head towards the closest enemy ant
 def greedy_step(game_state, ant_team):
     board = game_state.board
-    
-    # Create actions
+    enemy = ac.Cell.ANT_BLUE if ant_team == ac.Cell.ANT_RED else ac.Cell.ANT_RED
+
+    # Find food and enemies to optimize step
+    food_locations = []
+    enemies = []
+    for i in range(ac.ROW):
+        for j in range(ac.COL):
+            if board[i][j] == ac.Cell.FOOD:
+                food_locations.append((i,j))
+            elif board[i][j] == enemy:
+                enemies.append((i,j))
+
+    def dist(coord1, coord2):
+        return (abs(coord1[1]-coord2[1]) + abs(coord1[0]-coord2[0]))
+
+    def closest(coord1,locations):
+        return None if not locations else min(locations,key=lambda x:dist(coord1,x))
+
+    def get_action(at_coord, to_coord):
+        if at_coord[1]-to_coord[1] > 0:
+            return ac.Action.LEFT
+        elif at_coord[1]-to_coord[1] < 0:
+            return ac.Action.RIGHT
+        elif at_coord[0]-to_coord[0] > 0:
+            return ac.Action.UP
+        elif at_coord[0]-to_coord[0] < 0:
+            return ac.Action.DOWN
+        return ac.Action.NONE
+
+    # Get all steps possible from the at_coord
+    # Choose step that minimizes min_coord (food) and maximizes max_coord (enemy)
+    def min_risk_max_reward_step(at_coord, min_coord, max_coord, board, teammate_positions):
+        if min_coord == None: # if there is no food left just attack
```

Comment:
Instead of having two function definitions of `min_max_dist` could we have one function with the if statement inside? Also it seems like those nested functions don't really need to take `min_coord` and `max_coord` as inputs since the outer function has a definition of them and they don't change.